src/server/api/routers/posts.ts

```diff
@@ -1,9 +1,54 @@
+import { clerkClient } from "@clerk/nextjs";
+import type { User } from "@clerk/nextjs/dist/types/server";
+import { TRPCError } from "@trpc/server";
 import { z } from "zod";
 
 import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
 
+// custom function to filter user properties
+const filterUser = (user: User) => {
+  return {
+    id: user.id, 
+    username: user.username,
+    imageUrl: user.imageUrl
+  }
+}
+
 export const postRouter = createTRPCRouter({
-  getAll: publicProcedure.query(({ ctx }) => {
-    return ctx.db.post.findMany();
+  getAll: publicProcedure.query(async({ ctx }) => {
```

Comment:
```suggestion
  index: publicProcedure.query(async({ ctx }) => {
```

おそらくですが、この後Endpointが増えると思います。
Endpointを増やす前に命名規則について共有しておきますね！

SkygridではRuby Railsで使用していたEndpointの設計を採用しています

`getAll`のようにたくさんのデータを一度で取得するときは`index`を使用しています！
下記のリンクは日本語ですが参考のリンクです
こちらも難しいようであればハドル投げてきてください！

[Rails api ](https://qiita.com/k-penguin-sato/items/adba7a1a1ecc3582a9c9)